File: game-admin/src/main/java/com/dkm/updateImage/controllers/PictureController.java
package com.dkm.updateImage.controllers;

import com.dkm.config.annotations.CheckAdminPermission;
import com.dkm.updateImage.entities.ImageUpdateResultVo;
import com.dkm.updateImage.services.IPictureService;
import io.swagger.annotations.Api;
import io.swagger.annotations.ApiImplicitParam;
import io.swagger.annotations.ApiImplicitParams;
import io.swagger.annotations.ApiOperation;
import org.jetbrains.annotations.NotNull;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestPart;
import org.springframework.web.bind.annotation.RestController;
import org.springframework.web.multipart.MultipartFile;

import javax.annotation.Resource;
import javax.servlet.http.HttpServletRequest;

@Api("上传文件相关")
@RestController("/image/")

public class PictureController {
    @Resource
    private IPictureService pictureService;

    @ApiOperation("上传图片")
    @ApiImplicitParams({@ApiImplicitParam(
            dataType = "String",
            paramType = "header",
            required = true,
            value = "请求的Token",
            name = "TOKEN"
    )})
    @CheckAdminPermission
    @PostMapping(
            consumes = {"multipart/form-data"},
            produces = {"application/json;charset=UTF-8"},
            value = {"/update"}
    )
    @NotNull
    public ImageUpdateResultVo update(@RequestPart("file") @NotNull MultipartFile multipartFile, @NotNull HttpServletRequest httpServletRequest) {
        String token = httpServletRequest.getHeader("TOKEN");
        return pictureService.update(multipartFile, token);
    }
}

File: game-admin/src/main/java/com/dkm/updateImage/controller/PictureController.kt
package com.dkm.updateImage.controller

import com.dkm.config.annon.CheckAdminPermission
import com.dkm.updateImage.service.IPictureService
import io.swagger.annotations.ApiImplicitParam
import io.swagger.annotations.ApiImplicitParams
import io.swagger.annotations.ApiOperation
import org.springframework.data.repository.query.Param
import org.springframework.http.MediaType
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestPart
import org.springframework.web.bind.annotation.RestController
import org.springframework.web.multipart.MultipartFile
import javax.annotation.Resource


@RestController("/image/")
class PictureController {
    @Resource
    private lateinit var pictureService: IPictureService

    @ApiOperation("上传图片")
    @ApiImplicitParams(ApiImplicitParam(paramType = "header", name = "TOKEN", required = true, dataType = "String", value = "请求的Token"))
    @CheckAdminPermission
    @PostMapping("/update" ,consumes = [MediaType.MULTIPART_FORM_DATA_VALUE])
    fun update(@RequestPart("file") multipartFile: MultipartFile) = pictureService.update(multipartFile)
}